modin/data_management/factories.py

```diff
@@ -210,3 +210,42 @@ def prepare(cls):
         from modin.experimental.engines.pyarrow_on_ray.io import PyarrowOnRayIO
 
         cls.io_cls = PyarrowOnRayIO
+
+
+class ExperimentalPandasOnCloudrayFactory(ExperimentalBaseFactory):
+    @classmethod
+    def prepare(cls):
+        # query_compiler import is needed so remote PandasQueryCompiler has an imported local counterpart;
+        # if there isn't such counterpart rpyc generates some bogus class type which raises TypeError()
+        # upon checking its isinstance() or issubclass()
+        import modin.backends.pandas.query_compiler  # noqa: F401
```

Comment:
Importing the module is the fix, not the specific class type?

Reply:
yes, that's how RPyC works when constructing a netref (a "remote class") - it checks `sys.modules[]` looking for a module that holds remote class, so we actually don't need this module here but we need it imported before any `.read_*()` method (which returns a remote `QueryCompiler`) is called.